project/index.mjs

```diff
@@ -204,8 +204,135 @@ router.route("/:id/remove-member").post(auth0Middleware(), async (req, res) => {
+
+    if (!accessInfo.hasAccess) return respondWithError(res, 403, accessInfo.message)
+
+    const groupId = projectObj.data.group
+    const group = new Group(groupId)
+    group.setMemberRoles(collaboratorId, roles)
+
+    res.status(200).json({ message: `Roles [${roles}] updated for member ${collaboratorId}.` })
+  } catch (error) {
+    res.status(error.status || 500).json({ message: error.message || "Error updating member roles." })
+  }
+})
+
+
+// Remove a Role from Member
+router.route("/:projectId/collaborator/:collaboratorId/removeRoles").post(auth0Middleware(), async (req, res) => {
+  const { projectId, collaboratorId } = req.params
+  const roles = req.body.roles ?? req.body
+  const user = req.user
+  if (!user) {
+    return respondWithError(res, 401, "Unauthenticated request")
+  }
+  if (!roles) {
+    return respondWithError(res, 400, "Provide role(s) to remove")
+  }
+  if (roles.includes("OWNER")) {
+    return respondWithError(res, 400, "The OWNER role cannot be removed.")
+  }
+  try {
+    const projectObj = await new Project(projectId)
+    const accessInfo = await projectObj.checkUserAccess(user._id, ACTIONS.DELETE, SCOPES.ALL, ENTITIES.MEMBER)
+
+    if (!accessInfo.hasAccess) return respondWithError(res, 403, accessInfo.message)
+
+    const groupId = projectObj.data.group
+    const group = new Group(groupId)
+    await group.removeMemberRoles(collaboratorId, roles)
+
+    res.status(204).send()
+
+  } catch (error) {
+    res.status(error.status || 500).json({ message: error.message || "Error removing roles from member." })
+  }
+})
+
+
+// Switch project owner
+
+router.route("/:projectId/switch/owner").post(auth0Middleware(), async (req, res) => {
+  const { projectId } = req.params
+  const { newOwnerId } = req.body
+  const user = req.user
+
+  if (!user) {
+    return respondWithError(res, 401, "Unauthenticated request")
+  }
+  if (!newOwnerId) {
+    return respondWithError(res, 400, "Provide the ID of the new owner.")
+  }
+
+  try {
+    const projectObj = await new Project(projectId)
+
+    const accessInfo = await projectObj.checkUserAccess(user._id, ACTIONS.ALL, SCOPES.ALL, ENTITIES.ALL)
+    if (!accessInfo.hasAccess) return respondWithError(res, 403, "Only the current owner can transfer ownership.")
+
+    const groupId = projectObj.data.group
+    const group = new Group(groupId)
+
+    if (user._id === newOwnerId) {
+      return respondWithError(res, 400, "Cannot transfer ownership to the current owner.")
+    }
+
+    const currentRoles = await group.getMemberRoles(user._id)
+    // If user only has the OWNER role, we default them to CONTRIBUTOR before transferring ownership
+    Object.keys(currentRoles).length === 1 && await group.addMemberRoles(user._id, ["CONTRIBUTOR"])
+    await group.addMemberRoles(newOwnerId, ["OWNER"])
+    await group.removeMemberRoles(user._id, ["OWNER"])
+
+    res.status(200).json({ message: `Ownership successfully transferred to member ${newOwnerId}.` })
```

Comment:
Is there a compelling reason this needs to be JSON? Why not just a textual message as the body?